dataplane/src/pipeline/static_nf.rs

```diff
@@ -23,6 +23,13 @@ pub trait NetworkFunction<Buf: PacketBufferMut> {
         &'a mut self,
         input: Input,
     ) -> impl Iterator<Item = Packet<Buf>> + 'a;
+
+    /// Allow objects implementing [`NetworkFunction`] to identify themselves by overriding this.
+    #[allow(dead_code)]
+    #[allow(clippy::unnecessary_literal_bound)]
+    fn nf_name(&self) -> &str {
```

Comment:
I have no particular objection to the idea of an `nf_name` method.

That said, is the goal to make the `nf_name` a per-instance construct or a per implementation construct.

If the former, the I think this is basically correct.  If the later then I think you would want the return type to be `&'static str`

Reply:
It's meant to be per instance. I've been working on a setup that has two NFs of the same type chained and the name (which btw  should be unique (*)) should be able to distinguish both.

(*) = to ensure uniqueness, I'm considering that the add_nf() method or the like could actually return some form of Id that would ensure uniqueness; the code as it is now assumes the caller picks a distinct name per instance.